web/views/api/key_view.ex

```diff
@@ -5,12 +5,21 @@ defmodule HexWeb.API.KeyView do
     do: render_many(keys, __MODULE__, "key")
   def render("show." <> _, %{key: key}),
     do: render_one(key, __MODULE__, "key")
+  def render("delete." <> _, %{key: key}),
+    do: render_one(key, __MODULE__, "key")
 
   def render("key", %{key: key}) do
     entity =
-      key
-      |> Map.take([:name, :inserted_at, :updated_at])
-      |> Map.put(:url, key_url(HexWeb.Endpoint, :show, key))
+      if key.revoked_at do
+        %{
+          name: key.revoked_name,
+          revoked_at: key.revoked_at
+        }
+      else
+        key
+        |> Map.take([:name, :inserted_at, :updated_at])
+        |> Map.put(:url, key_url(HexWeb.Endpoint, :show, key))
+      end
```

Comment:
Good point, just special case the to make it `nil` since that route will 404 from now on.